actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -604,10 +604,16 @@ def apply_form_for_options!(record, object, options) #:nodoc:
       #   This is helpful when fragment-caching the form. Remote forms
       #   get the authenticity token from the <tt>meta</tt> tag, so embedding is
       #   unnecessary unless you support browsers without JavaScript.
-      # * <tt>:local</tt> - By default form submits via typical HTTP requests.
-      #   Enable remote and unobtrusive XHRs submits with <tt>local: false</tt>.
-      #   Remote forms may be enabled by default by setting
-      #   <tt>config.action_view.form_with_generates_remote_forms = true</tt>.
+      # * <tt>:local</tt> - Whether to use standard HTTP form submission.
+      #   When set to <tt>true</tt>, the form is submitted via standard HTTP.
+      #   When set to <tt>false</tt>, the form is submitted as a "remote form", which 
+      #   is handled by Rails UJS as an XHR. When unspecified, the behavior is derived 
+      #   from <tt>config.action_view.form_with_generates_remote_forms</tt> where the 
+      #   config's value is actually the inverse of what <tt>local</tt>'s value would be.
+      #   As of Rails 6.1, that configuration option defaults to <tt>false</tt>
+      #   (which has the equivalent effect of passing <tt>local: true</tt>).
+      #   In previous versions of Rails, that configuration option defaults to
+      #   <tt>true</tt> (the equivalent of passing <tt>local: false</tt>).
```

Comment:
What if we just side-stepped the back-and-forth:

```suggestion
      #   is handled by Rails UJS as an XHR. Defaults to the inverse of 
      #   <tt>config.action_view.form_with_generates_remote_forms</tt>.
```

If a developer is unsure of the value of `config.action_view.form_with_generates_remote_forms` in their app, they can check it in the Rails console.  (Which might be the surest course of action, given its history.)

Reply:
I originally wrote it that way, but after asking some colleagues the feedback I got was that it wasn't clear enough. I agree that it seems like a lot of extra verbiage in essentially the wrong place—this configuration option ought to be documented elsewhere, not here.

But I disagree that they should check the console to see. I feel like a configuration option like this should be documented. Finding out the current value isn't the only action a developer might take, they should also be given an understanding of how to change it. The confusion we encountered here actually arose because we were upgrading an application, so we were in the position of choosing the new value and needed to understand what we wanted the value to be. In other words, should we adopt the 6.1 defaults?

So this text also serves to explain the interaction of this option and how the default value is based on your current version of Rails. I would much rather leave a developer understanding that interaction rather than coming away with instructions to check using the console.

Reply:
> I feel like a configuration option like this should be documented. Finding out the current value isn't the only action a developer might take, they should also be given an understanding of how to change it. The confusion we encountered here actually arose because we were upgrading an application, so we were in the position of choosing the new value and needed to understand what we wanted the value to be. In other words, should we adopt the 6.1 defaults?

In that case, what if we elaborate on the description of `config.action_view.form_with_generates_remote_forms` [in the Configuring guide](https://github.com/rails/rails/blame/73c0266fd61c266abe9e7c2ff4da41006740aa90/guides/source/configuring.md#L727)?  (Keep in mind: the historical default values are covered in the [6.1 defaults section](https://github.com/rails/rails/blame/73c0266fd61c266abe9e7c2ff4da41006740aa90/guides/source/configuring.md#L1073) and [5.1 defaults section](https://github.com/rails/rails/blame/73c0266fd61c266abe9e7c2ff4da41006740aa90/guides/source/configuring.md#L1099).)